tests/http_spec.rs

```diff
@@ -0,0 +1,216 @@
+        method_match && url_match
+      })
+      .expect("Mock not found");
+    // Assert upstream request
+    let upstream_request = mock.0.clone();
+    assert!(
+      self.expected_upstream_requests.contains(&upstream_request),
+      "Unexpected upstream request: {:?}",
+      upstream_request
+    );
+
+    // Clone the response from the mock to avoid borrowing issues.
+    let mock_response = mock.1.clone();
+
+    // Build the response with the status code from the mock.
+    let status_code = reqwest::StatusCode::from_u16(mock_response.0.status)?;
+    let mut response = Response { status: status_code, ..Default::default() };
+
+    // Insert headers from the mock into the response.
+    for (key, value) in mock_response.0.headers {
+      let header_name = HeaderName::from_str(&key)?;
+      let header_value = HeaderValue::from_str(&value)?;
+      response.headers.insert(header_name, header_value);
+    }
+
+    // Set the body of the response.
+    response.body = ConstValue::try_from(serde_json::from_value::<Value>(mock_response.0.body)?)?;
+
+    Ok(response)
+  }
+}
+
+#[tokio::test]
+async fn test_body_yaml() {
+  let spec = HttpSpec::read("tests/data/sample.yaml").unwrap();
+  assert_downstream(spec).await;
+}
+
+async fn assert_downstream(spec: HttpSpec) {
+  let has_only_annotation = spec
+    .downstream_assertions
+    .iter()
+    .any(|assertion| matches!(assertion.annotation, Some(Annotation::Only)));
+
+  for downstream_assertion in spec.downstream_assertions.iter() {
+    match &downstream_assertion.annotation {
+      Some(Annotation::Skip) if !has_only_annotation => {
+        let request_details = format_request_details(&downstream_assertion.request.0);
+        println!("Skipping test in: {}\nRequest Details: {}", spec.name, request_details);
+        continue;
+      }
+      Some(Annotation::Only) | None
+        if !has_only_annotation || matches!(downstream_assertion.annotation, Some(Annotation::Only)) =>
+      {
+        let response = run(spec.clone(), &downstream_assertion).await.unwrap();
+        let body = hyper::body::to_bytes(response.into_body()).await.unwrap();
+        assert_eq!(
+          body,
+          serde_json::to_string(&downstream_assertion.response.0.body).unwrap()
+        );
+      }
+      Some(Annotation::Fail) if !has_only_annotation => {
+        let response = run(spec.clone(), &downstream_assertion).await;
+        assert!(response.is_err());
+      }
+      _ => {} // Skip other cases if "Only" is present in any assertion
+    }
+  }
+}
+
+// Helper function to format request details for printing.
+fn format_request_details(request: &APIRequest) -> String {
+  format!(
+    "Method: {:?}, Path: {}, Body: {}",
+    request.method.clone(),
+    request.url.clone(),
+    request.body.clone()
+  )
+}
+
+#[tokio::test]
+async fn test_body_json() {
+  let spec = HttpSpec::read("tests/data/sample.json").unwrap();
```

Comment:
```suggestion
  let spec = HttpSpec::read("tests/http/sample.json").unwrap();
```

Like `/graphql` folder, let's keep `/http` folder.

Reply:
Secondly, we need to load all the files at once. We shouldn't need to update the spec code every time we add a new file.

Reply:
done